src/realm_dart_sync.cpp

```diff
@@ -22,6 +22,32 @@
 #include "realm_dart.hpp"
 #include "realm_dart_sync.h"
 
+class WaitingCallback
```

Comment:
Why do we need this class?

Reply:
I wanted to create a method that is not visible in the external realm.h API. Could be better solution.

Reply:
A freestanding function would not be visible in realm.h.

Reply:
I wanted to say realm_dart_sync.h, not realm.h. Anyway, my mistake probably was that I had defined the function at the end of the file. If it is defined before to be called it builds and it is not needed to be exposed in realm_dart_sync.h. Thanks, I will remove the class.